opentelemetry-instrumentation-tower/README.md

```diff
@@ -0,0 +1,208 @@
+# Tower OTEL Metrics Middleware
+
+OpenTelemetry Metrics Middleware for Tower-compatible Rust HTTP servers.
+
+## Examples
+
+See `examples` directory in repo for runnable code and supporting config files.
+Attempts are made to keep the code here synced, but it will not be perfect.
+
+OTEL libraries in particular are sensitive to minor version changes at this point,
+so the examples may only work with the OTEL crate versions pinned in `examples`.
+
+### Axum Server
+
+Adding OpenTelementry HTTP Server metrics using the [`Axum`](https://docs.rs/axum/latest/axum) framework
+over a Tower-compatible [`Hyper`](https://docs.rs/hyper/latest/hyper) Service:
+
+```rust
+use axum::routing::{get, post, put, Router};
+use bytes::Bytes;
+use opentelemetry::global;
+use opentelemetry_otlp::{
+    WithExportConfig, {self},
+};
+use opentelemetry_sdk::metrics::PeriodicReader;
+use opentelemetry_sdk::Resource;
+use std::time::Duration;
+use otel_tower_metrics;
+
+const SERVICE_NAME: &str = "example-axum-http-service";
+// Metric export interval should be less than or equal to 15s
+// if the metrics may be converted to Prometheus metrics.
+// Prometheus' query engine and compatible implementations
+// require ~4 data points / interval for range queries,
+// so queries ranging over 1m requre <= 15s scrape intervals.
+// OTEL SDKS also respect the env var `OTEL_METRIC_EXPORT_INTERVAL` (no underscore prefix).
+const _OTEL_METRIC_EXPORT_INTERVAL: Duration = Duration::from_secs(10);
+
+fn init_otel_resource() -> Resource {
+    Resource::builder().with_service_name(SERVICE_NAME).build()
+}
+
+// PCT_SLOW_REQUESTS and MAX_SLOW_REQUEST_SEC are used to inject latency into some responses
+// in order to utilize the higher request duration buckets in the request duration histogram.
+// These values are chosen so that with the load-gen script's max 100 VUs, we get just enough
+// slow requests to show up on the histograms without completely blocking up the server.
+const PCT_SLOW_REQUESTS: u64 = 5;
+const MAX_SLOW_REQUEST_SEC: u64 = 16;
+// MAX_BODY_SIZE_MULTIPLE is used to demonstrate the `http.server.response.body.size` histogram
+const MAX_BODY_SIZE_MULTIPLE: u64 = 16;
+
+#[axum::debug_handler]
+async fn handle() -> Bytes {
+    if rand_09::random_range(0..100) < PCT_SLOW_REQUESTS {
+        let slow_request_secs = rand_09::random_range(0..=MAX_SLOW_REQUEST_SEC);
+        tokio::time::sleep(Duration::from_secs(slow_request_secs)).await;
+    };
+    let body_size_multiple = rand_09::random_range(0..=MAX_BODY_SIZE_MULTIPLE);
+    Bytes::from("hello world\n".repeat(body_size_multiple as usize))
+}
+
+#[tokio::main]
+async fn main() {
+    let exporter = opentelemetry_otlp::MetricExporter::builder()
```

Comment:
for instrumentation's docs, it's easy to use the stdout exporter. It'll be hard to maintain proper OTLP example for each instrumentations.